packages/app-store/salesforce/lib/CrmService.ts

```diff
@@ -323,108 +323,190 @@ export default class SalesforceCRMService implements CRM {
     includeOwner?: boolean;
     forRoundRobinSkip?: boolean;
   }) {
-    const conn = await this.conn;
-    const emailArray = Array.isArray(emails) ? emails : [emails];
-    const appOptions = this.getAppOptions();
-    const recordToSearch =
-      (forRoundRobinSkip ? appOptions?.roundRobinSkipCheckRecordOn : appOptions?.createEventOn) ??
-      SalesforceRecordEnum.CONTACT;
-    let soql: string;
-    let accountOwnerId = "";
-    if (recordToSearch === SalesforceRecordEnum.ACCOUNT) {
-      // For an account let's assume that the first email is the one we should be querying against
-      const attendeeEmail = emailArray[0];
-      soql = `SELECT Id, Email, OwnerId, AccountId FROM Contact WHERE Email = '${attendeeEmail}' AND AccountId != null`;
-
-      // If this is for a round robin skip then we need to return the account record
-      if (forRoundRobinSkip) {
-        const results = await conn.query(soql);
-        if (results.records.length) {
-          const contact = results.records[0] as { AccountId: string };
-          if (contact) {
-            soql = `SELECT Id, OwnerId FROM Account WHERE Id = '${contact.AccountId}'`;
-          }
-        } else {
-          // If we can't find the exact contact, then we need to search for an account where the contacts share the same email domain
-          const accountId = await this.getAccountIdBasedOnEmailDomainOfContacts(attendeeEmail);
-          if (accountId) {
-            soql = `SELECT Id, OwnerId FROM Account WHERE Id = '${accountId}'`;
+    const log = logger.getSubLogger({ prefix: [`[getContacts]:${emails}`] });
+    try {
```

Comment:
As a followup, we should add try-catch at a higher level so that no CRM app can crash the booking page and not just salesforce.

Reply:
AFAIK CrmManager prevents a crash to block bookings doesn't it?

Reply:
This fn is called while loading the booking page. There it isn't handled